src/main/java/cubicchunks/lighting/FirstLightProcessor.java

```diff
@@ -74,6 +75,18 @@ public FirstLightProcessor(GeneratorStage lighting, String name, ICubeCache cach
 		ICubicWorld world = cube.getWorld();
 		ICubeCache cache = world.getCubeCache();
 
+		for (int x = -2; x <= 2; ++x) {
+			for (int y = -2; y <= 2; ++y) {
+				for (int z = -2; z <= 2; ++z) {
+					CubeCoords coords = new CubeCoords(cube.getX() + x, cube.getY() + y, cube.getZ() + z);
+					if (!cache.cubeExists(coords) || cache.getCube(coords).getCurrentStage().precedes(generatorStage)) {
+						CubicChunks.LOGGER.error("Missing cube ({})! {}/{}", coords, cache.getCube(coords).getCurrentStage().getName(), generatorStage.getName());
+						System.exit(1);
```

Comment:
1. NEVER use System.exit(). Use `FMLCommonHandler.exitJava()`. Or throw `new Error().`
2. You can simply create `new  FastCubeBlockAccess(cubeCache, centerCube, radius)` and modify it to also check stage - it will throw exception if required cube isn't loaded and it will dump information about which cubes are loaded (you can also modify that to include stage if it doesn't do that now)